internal/web/api/api.go

```diff
@@ -66,7 +66,12 @@ func listComponentsHandlerRemoteCfg(host service.Host) http.HandlerFunc {
 			return
 		}
 
-		listComponentsHandlerInternal(svc.Data().(remotecfg.Data).Host, w, r)
+		data := svc.Data().(remotecfg.Data)
+		if data.Host == nil {
+			http.Error(w, "remote config service startup in progress", http.StatusInternalServerError)
+			return
+		}
+		listComponentsHandlerInternal(data.Host, w, r)
```

Comment:
should we have the same check in getComponentHandlerRemoteCfg? It may no be possible to trigger via the UI but could be in theory by sending http requests directly

Reply:
Definitely, great catch.